### ISSUE TITLE
v5 event package

### ISSUE THREAD

**Issue by hunyadvari-peter** [go-micro/go-micro#2728]
The event package has been deleted in v5 branch.
Is there any plan to do this? 
How can we replace the event handling? 
Brokers have much less functionality, and in theory no guarantee of delivery.

**Comment by asim** [comment]
I think the two interfaces had a lot of overlapping concerns. So we could add some of that to the broker.  Can you detail what parts you'd like to include?

**Comment by hunyadvari-peter** [comment]
I agree with you, but the two interfaces had different purposes. Broker = pub/sub, event = event streaming with multiple policies.
Options what is missing from broker:
AckWait time.Duration
RetryLimit int
AutoAck bool
CustomRetries bool
...

From message the Ack,Nack functions.

And the most important think is to handle topic filtering, with the broker is not possible.
Eg.: unable to subscribe to topic.*

**Comment by asim** [comment]
What's your ideal solution? Reintroduce the event interface?

**Comment by asim** [comment]
If you need it. You can make a PR to bring it back.

**Comment by vtolstov** [comment]
Broker can do all the things, auto ack is simple if defer func() { 
if err == nil {
  return evt.Ack()
}
or can be handled in broker implementation. Topic filtering not the problem - in kafka for example you can subscribe via regexp to topics

**Comment by hunyadvari-peter** [comment]
> Topic filtering not the problem - in kafka for example you can subscribe via regexp to topics

We will bring a solution in PR


> autoack

It is not a good solution in some situation.

> Topic filtering not the problem - in kafka for example you can subscribe via regexp to topics

I do not know nothing about kafka, but a lot of event broker use a * and similar character for filtering in a topic name. But rpc_server and the router try to find subscriber with the topic name he got from the broker, which is the full topic name, not the filtered. eg.:
we subscribe to this: topic.customer_*
we got event from this topic: topic.customer_micro

But we will bring a solution for this "problem" too, in PR.

**Comment by vtolstov** [comment]
ah =) in my fork i'm remove ability to subscribe via server package and use broker directly, with ability to unmarshal to needed struct pointer

**Comment by asim** [comment]
@hunyadvari-peter do we still need the v5 event package? If so we'll reintroduce.

**Comment by asim** [comment]
@hunyadvari-peter if you think this is going to be useful I will merge it https://github.com/micro/go-micro/pull/2761

**Comment by asim** [comment]
We brought back the package
